src/canvas/graphic.ts

```diff
@@ -20,9 +20,17 @@ import { REDARAW_BIT, SHAPE_CHANGED_BIT } from '../graphic/constants';
 const pathProxyForDraw = new PathProxy(true);
 
 // Not use el#hasStroke because style may be different.
-function styleHasStroke(style: PathStyleProps, excludeLineWidth?: boolean) {
+function styleHasStroke(style: PathStyleProps) {
     const stroke = style.stroke;
-    return stroke != null && stroke !== 'none' && (excludeLineWidth || style.lineWidth > 0);
+    return !(stroke == null || stroke === 'none' || !(style.lineWidth > 0));
+}
+
+// ignore lineWidth and must be string
+// Expected color but found '[' when color is gradient
+function isValidStrokeFillStyle(
+    strokeOrFill: PathStyleProps['stroke'] | PathStyleProps['fill']
+): strokeOrFill is string {
+    return strokeOrFill != null && strokeOrFill !== 'none' && typeof strokeOrFill === 'string';
```

Comment:
I think we can ignore the `string` check. It won't happen only if the user sets a non-string `color`. But in this case, the check can be very complex, for example, the string is not a valid color string.

We can stop at eliminating the warning from ourself code.